src/main/java/org/eclipse/yasson/internal/model/ReverseTreeMap.java

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright (c) 2015, 2019 Oracle and/or its affiliates. All rights reserved.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v1.0 and Eclipse Distribution License v. 1.0
+ * which accompanies this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * Contributors:
+ * Sebastien Rius
+ ******************************************************************************/
+package org.eclipse.yasson.internal.model;
+
+import java.util.Comparator;
+import java.util.TreeMap;
+
+/**
+ * TreeMap with a reverse ordering by default
+ * 
+ * @author Sebastien Rius
+ * 
+ */
+public class ReverseTreeMap<K extends Comparable<? super K>, V> extends TreeMap<K, V> {
```

Comment:
Is the `<K extends Comparable<? super K>, V>` parameterization necessary here?

I would think we can just do `public class ReverseTreeMap extends TreeMap<K, V>`

Reply:
The only way to remove parametrization, is to remove it completely : 
`public class ReverseTreeMap extends TreeMap`
but it leads to java warning about raw types not being parameterized...
If we parameterize the class, you need to specify `Comparable` extension in order for `super` constructor to be found.
Do you prefer removing parametrization with `@SuppressWarnings` ?

Reply:
ok, thanks for checking. I think it's fine the way it is